### PR TITLE
 Add SmallGroupsAvailable, NumberSmallGroupsAvailable, IdGroupsAvailable 

### LINKED ISSUE #20: Add `SmallGroupsAvailable`, `NumerSmallGroupsAvailable`, `IdGroupsAvailable`
The `transgrp` and `primgrp` packages offer user functions `PrimitiveGroupsAvailable` and `TransitiveGroupsAvailable` to make it possible to query what data is available. It would be nice if `smallgrp` offered something similar.

Of course there are `SMALL_AVAILABLE` and `ID_AVAILABLE`, but they are undocumented internal functions, and as such their interface is not really straight-forward or user friendly. I suggest adding these:
* `SmallGroupsAvailable(order)` returning true/false if a list of all small groups of that order is available
* `NumberSmallGroupsAvailable(order)` --  almost the same, but returns `true` for 1024.
* `IdGroupsAvailable(order)`

All of them should be straight forward to implement using `{SMALL,ID}_AVAILABLE`. I'd whip up a patch, but then you'd rightfully ask for tests etc.... anyway, didn't want this to be lost.

Of course we then could wonder whether to add them to GAP, too, for the transition, similar to what was done for `transgrp` and `primgrp`. But perhaps that is simply not necessary, as nothing in the library needs this right now, I think...

### PR DESCRIPTION
**Pull request by markuspf**
Resolves #20

### CODE REVIEW

File: gap/small.gd [hunk 1]
```diff
@@ -336,6 +370,23 @@ UnbindGlobal( "ID_AVAILABLE" );
...
+#############################################################################
+##
+#F  IdGroupsAvailable( <order> )
+##
+##  <#GAPDoc Label="IdGroupsAvailable">
+##  <ManSection>
+##  <Func Name="IdGroupsAvailable" Arg='order'/>
+##
+##  <Description>
+##  returns <C>true</C>, if the identification routines for of groups of
```
Comment (wilfwilson) [inline gap/small.gd]: Typo: "for of"

File: gap/small.gd [hunk 2]
```diff
@@ -80,6 +80,40 @@ UnbindGlobal( "SMALL_AVAILABLE" );
...
+#############################################################################
+##
+#F  SmallGroupsAvailable( <order> )
+##
+##  <#GAPDoc Label="SmallGroupsAvailable">
+##  <ManSection>
+##  <Func Name="SmallGroupsAvailable" Arg='order'/>
+##
+##  <Description>
+##  returns <C>true</C> if the library of groups of <A>order</A> is installed, and
```
Comment (wilfwilson) [inline gap/small.gd]: In my opinion this sounds strange: substituting 6 for `<order>`, we get `"...if the library of groups of 6 is installed"`. The same is true for the other bits of documentation. The solution would be to write `"...if the library of groups of order <A>order</A> is installed"`, which I realise is repetitive. So I wouldn't complain if you kept it how it is – I think it is suitably clear.

**Comment by markuspf** [comment]
Thanks for the review @wilfwilson, I addressed your remarks and will wait for the tests to cycle before merging.

File: gap/small.gd [hunk 3]
```diff
@@ -72,14 +72,48 @@ UnbindGlobal( "SMALL_AVAILABLE" );
...
+##  installed, and <C>false</C> otherwise.
+##  </Description>
+##  </ManSection>
+##  <#/GAPDoc>
+##
+DeclareGlobalFunction( "SmallGroupsAvailable" );
+
+#############################################################################
+##
+#F  NumberSmallGroupsAvailable( <order> )
```
Comment (fingolfin) [inline gap/small.gd]: I wonder if this (and the other ones) shouldn't perhaps instead be called `HasNumberSmallGroups`?

File: gap/small.gd [hunk 4]
```diff
@@ -72,14 +72,48 @@ UnbindGlobal( "SMALL_AVAILABLE" );
...
+##
+##  <#GAPDoc Label="NumberSmallGroupsAvailable">
+##  <ManSection>
+##  <Func Name="NumberSmallGroupsAvailable" Arg='order'/>
+##
+##  <Description>
+##  returns <C>true</C> if the number of groups of order <A>order</A> is known, and
+##  <C>false</C> otherwise.
```
Comment (fingolfin) [inline gap/small.gd]: Perhaps make it crystal clear that this affects whether `NumberSmallGroups(order)` will return a value or run into an error? E.g. add "In the former case, the number can be retrieved using `<Ref Func="NumberSmallGroups"/>.`

Also, a short example could illuminate this:
```
gap> NumberSmallGroupsAvailable(100);
gap> NumberSmallGroups(100);
16
gap> NumberSmallGroupsAvailable(2048);
false
gap> NumberSmallGroups(2048);
Error, FOOBAR
```

File: gap/small.gd [hunk 5]
```diff
@@ -328,14 +362,31 @@ UnbindGlobal( "ID_AVAILABLE" );
...
 ##  </Description>
 ##  </ManSection>
 ##
 DeclareGlobalFunction( "ID_AVAILABLE" );
 
+#############################################################################
+##
+#F  IdGroupsAvailable( <order> )
+##
+##  <#GAPDoc Label="IdGroupsAvailable">
```
Comment (fingolfin) [inline gap/small.gd]: Hmm, maybe `IdGroupAvailable` or `HasIdGroup`, without the "s", to match `IsGroup`?

File: gap/small.gi [hunk 6]
```diff
@@ -37,6 +37,20 @@ InstallGlobalFunction( SMALL_AVAILABLE, function( size )
...
+end);
+
+InstallGlobalFunction( NumberSmallGroupsAvailable,
+function(order)
+    if order = 1024 then
+        return true;
+    else
+        return SmallGroupsAvailable(order);
+    fi;
+end);
```
Comment (fingolfin) [inline gap/small.gi]: I plan an extension of the SGL which adds many group orders for which no list of groups is available. For this, this function will need to be changed (e.g. by turning it into an operation). But that can be done later, of course.

File: tst/small.tst [hunk 7]
```diff
@@ -30,6 +30,11 @@ gap> notavailable;
   3726, 3744, 3750, 3752, 3760, 3768, 3780, 3784, 3792, 3800, 3808, 3816, 
   3834, 3840, 3848, 3861, 3864, 3872, 3880, 3888, 3912, 3915, 3920, 3936, 
   3942, 3944, 3952, 3960, 3969, 3976, 3984, 3993, 3996, 4000 ]
+gap> notavailable2 := Filtered([1..4000], x -> not SmallGroupsAvailable(x));;
+gap> notavailable = notavailable2;
+true
+gap> Filtered([1..4000], i -> NumberSmallGroupsAvailable(i) <> SmallGroupsAvailable(i));
+[ 1024 ]
```
Comment (fingolfin) [inline tst/small.tst]: This test *will* break with future extensions of the SGL. Perhaps restrict from 4000 to 2000, the range which the SGL actually promises?

File: tst/small.tst [hunk 8]
```diff
@@ -202,6 +207,9 @@ gap> notavailable;
   4752, 4760, 4776, 4784, 4800, 4806, 4816, 4824, 4840, 4848, 4860, 4872, 
   4875, 4880, 4888, 4896, 4914, 4920, 4928, 4944, 4960, 4968, 4984, 4992, 
   4995, 5000 ]
+gap> notavailable2 := Filtered([1..5000], x -> not IdGroupsAvailable(x));;
+gap> notavailable = notavailable2;
+true
```
Comment (fingolfin) [inline tst/small.tst]: Same as above.

File: gap/small.gd (hunk 5 above)
Comment (alex-konovalov) [reply gap/small.gd]: `HasIdGroup` looks like a tester for an attribute - IMHO `IdGroupAvailable` is more consistent and also follows similar conventions elsewhere.

File: gap/small.gi (hunk 6 above)
Comment (fingolfin) [reply gap/small.gi]: While I stand by this, of course also `NumberSmallGroups` hardcodes the 1024, so I guess more changes will be needed. No reason to block this PR over that.

File: gap/small.gd (hunk 5 above)
Comment (fingolfin) [reply gap/small.gd]: Actually, upon re-reading this, I think `IdGroupsAvailable` (with an "s") is available, as IdGroup applies to a specific group, and this is about whether the id of all group*s* of given order are available.

**Pull request merged**